### PR TITLE
Skip strict check on PKCE discovery claim on Azure

### PR DESCRIPTION
**Pull request by bobcallaway**
This skips the check for the `code_challenge_methods_supported` claim
within the OIDC discovery document when configured to connect to an
Azure-based OIDC provider. AFAIK, Azure OIDC does in fact support PKCE
but does not advertise this in its discovery document so this should
enable additional use cases with `cosign` and Azure.

Partially-Fixes: https://github.com/sigstore/cosign/issues/1354

Signed-off-by: Bob Callaway <bob.callaway@gmail.com>

### CODE REVIEW

**Comment by jw-s** [comment]
@bobcallaway I'll build this branch tomorrow and test it against Azure if you need this verification?

**Comment by bobcallaway** [comment]
> @bobcallaway I'll build this branch tomorrow and test it against Azure if you need this verification?

please, that would be awesome

**Review by dlorenc — DISMISSED** [review]

File: pkg/oauthflow/pkce.go [hunk 1]
```diff
@@ -57,7 +58,7 @@ func NewPKCE(provider *oidc.Provider) (*PKCE, error) {
 			fmt.Printf("Unsupported code challenge method in list: '%v'", method)
 		}
 	}
-	if chosenMethod == "" {
+	if chosenMethod == "" && !providerIsAzureBacked(provider) {
```
Comment (jw-s) [inline pkg/oauthflow/pkce.go:61]: This means we pass an empty string as the `code_challenge_method` query parameter value which is wrong. This needs to be set to `S256` as per RFC7636 it should support it.

**Comment by bobcallaway** [comment]
@jw-s could you take another look at the updated PR when you have a chance? Thanks!

**Comment by jw-s** [comment]
@bobcallaway  Sorry for the delay - It works as expected now, tested via the interactive flow with confidential client (client secret)

**Comment by bobcallaway** [comment]
> @bobcallaway Sorry for the delay - It works as expected now, tested via the interactive flow with confidential client (client secret)

Thanks for verifying this!

**Review by dlorenc — APPROVED** [review]

**Pull request merged**